scripts/mainnet_create_markets.js

```diff
@@ -0,0 +1,156 @@
+  const marketImpl = await Market.deployed()
+  const marketRegistryImpl = await MarketsRegistry.deployed()
+  const ammImpl = await MinterAmm.deployed()
+
+  // https://etherscan.io/token/0x2260fac5e5542a773aa44fbcfedf7c193bc2c599
+  const WBTCToken = await IERC20.at(
+    "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599",
+  )
+
+  // https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48
+  const USDCToken = await IERC20.at(
+    "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+  )
+
+  console.log(`Setting up proxy pointing at the market registry logic`)
+  const proxyContract = await Proxy.new(marketRegistryImpl.address)
+  console.log(`Initializing MarketsRegistry`)
+  const deployedMarketsRegistry = await MarketsRegistry.at(
+    proxyContract.address,
+  )
+
+  const result = await deployedMarketsRegistry.initialize(
+    tokenImpl.address,
+    marketImpl.address,
+    ammImpl.address,
+  )
+
+  console.log(
+    `Market registry is up and running at address ${deployedMarketsRegistry.address}, initialized with tx id ${result.tx}`,
+  )
+
+  let priceOracle
+  if (network == "mainnet") {
+    console.log("connecting to mainnet oracle")
+    // see https://etherscan.io/address/0xF4030086522a5bEEa4988F8cA5B36dbC97BeE88c
+    priceOracle = await AggregatorV3Interface.at(
+      "0xF4030086522a5bEEa4988F8cA5B36dbC97BeE88c",
+    )
+  } else if (network == "kovan") {
+    console.log("connecting to kovan oracle")
+    // see https://kovan.etherscan.io/address/0x6135b13325bfC4B00278B4abC5e20bbce2D6580e
+    priceOracle = await AggregatorV3Interface.at(
+      "0x6135b13325bfC4B00278B4abC5e20bbce2D6580e",
+    )
+  } else if (network == "rinkeby") {
+    console.log("connecting to rinkeby oracle")
+    // https://rinkeby.etherscan.io/address/0xECe365B379E1dD183B20fc5f022230C044d51404
+    priceOracle = await AggregatorV3Interface.at(
+      "0xECe365B379E1dD183B20fc5f022230C044d51404",
+    )
+  } else if (network == "development") {
+    console.log("deploying price oracle")
+    priceOracle = await MockPriceOracle.deployed() // 8 decimals for WBTC's 8 decimals
+    await priceOracle.setLatestAnswer(14_000 * 10 ** 8) // hardcode to $14000
+    console.log("completed price oracle deploy")
+  }
+
+  // wait for 10 seconds for infura to catchup
+  console.log("waiting for infura to catchup on latest deployments...")
+  await delay(10 * 1000)
+
+  console.log("deploying 2 AMMs, one for WBTC-USDC and another for USDC-WBTC")
+  let ammWBTCUSDC
+  let ammUSDCWBTC
+  let ret = await deployedMarketsRegistry.createAmm(
+    priceOracle.address,
+    USDCToken.address,
+    WBTCToken.address,
+    0,
+    false,
+  )
+  let ammAddress = ret.logs[2].args["0"]
+  ammWBTCUSDC = await MinterAmm.at(ammAddress)
+
+  // set deposit limits and whitelist LPs for first round
+  await ammWBTCUSDC.setEnforceDepositLimits(true, "62440000") // 0.6244 BTC ~ $10K
+  await ammUSDCWBTC.setEnforceDepositLimits(true, "10000000000") // $10K
```

Comment:
This should return an error since the value of `ammUSDCWBTC` is being set later in the script

Reply:
good catch, fixed